### DISCUSSION TITLE
Distinguish whether the `body` of a message is HTML or plain text?

### DISCUSSION THREAD

**Discussion by kostrykin** [ecederstrand/exchangelib#1335] [Q&A]
To my knowledge, the body of an email can be HTML or plain text. Is it possible to distinguish whether the `body` of a message is HTML or plain text?

According to the docs https://ecederstrand.github.io/exchangelib/exchangelib/items/index.html#exchangelib.items.Item, items have the two attributes `body` and `text_body`. Unfortunately, these aren't documented. What is their difference?

**Comment by ecederstrand** [comment]
All classes refer to their MSDN documentation for further details. The docs you linked to has a reference to https://docs.microsoft.com/en-us/exchange/client-developer/web-service-reference/item where you can see a description of the two attributes. In exchangelib, the field returns objects of type `exchangelib.properties.Body` or `exchangelib.properties.HTMLBody` depending on what the server returns.
